packages/relay-compiler/codegen/RelayFileWriter.js

```diff
@@ -265,11 +266,21 @@ class RelayFileWriter implements FileWriterInterface {
               node.name,
             );
 
+            const recursiveFields = (flowNode.argumentDefinitions || [])
+              .map(arg =>
+                getFieldNameSCCs(arg.type).filter(
```

Comment:
I was wondering if we need to calculate the `recursiveFields` ahead of time? I wonder if as a simplified solution we can just use the recursion limit when recursing inside `transformNonNullableInputType`?
I might be missing something obvious, but I'm not sure why we need this list of `recursiveFields` to achieve the same result.
It would be nice if we can avoid this to avoid the additional work and additional complexity in the code.

Reply:
@jstejada Thanks for looking at this PR!
The reason I opted to compute the recursive field names ahead of time was because I did not want to limit the depth of flow types generated for non-recursive types. If you only rely on a recursion limit you will cut off the generation of flow types for arbitrarily nested GraphQL types even if they are not recursive.
The reason I did not detect recursive types dynamically by keeping a record of what types had been seen during the depth first traversal is that I did not want to limit the flow type generation for GraphQL types that included two fields of the same recursive  type. The second field to be visited would not get the same flow type as its sibling.
And the reason I did not just create separate flow types for each GraphQL type and then simply refer to them in the generated flow type is because of a commit message I saw while digging through the issue that said this approach led to naming conflicts.
It would be helpful if you had further insight into any of these approaches. But so far the solution in this PR is the cleanest and most correct I could come up with.

Reply:
Ah, I see. Thanks for the clarification @benjaminogles! Your explanation sounds reasonable. I think the ideal solution would be 3), where we just generate recursive flow types; would you mind pointing me to the comment about the naming conflicts?

Other than that, before I can approve this, I need to be able to test it internally. Would you mind resolving existing merge conflicts?

Thank you!

Reply:
No problem. The commit I looked at is here https://github.com/facebook/relay/commit/d94380a9be64f2a72828effdb0c934ef9bf74aff

Reply:
hey @benjaminogles sorry for the delay on this PR. I think creating separate flow types for the input object types would be ideal; the only concern I have for this approach is how much it could affect runtime of the compiler internally. I've gone ahead and started some changes on top of this PR to try to generate separate flow types, unless you're interested in taking a stab at that approach.

Let me know what you think, thanks again for doing this!